hypothesis-python/src/hypothesis/extra/array_api.py

```diff
@@ -563,15 +617,25 @@ def _boolean_dtypes(xp: Any) -> st.SearchStrategy[DataType]:
         ) from None
 
 
-def _numeric_dtypes(xp: Any) -> st.SearchStrategy[DataType]:
-    """Return a strategy for all numeric dtype objects."""
+def _real_dtypes(xp: Any) -> st.SearchStrategy[DataType]:
+    """Return a strategy for all real dtype objects."""
```

Comment:
```suggestion
    """Return a strategy for all real-valued dtype objects."""
```

To make it clear that we're excluding complex numbers, not e.g. mock dtype objects.